### ISSUE TITLE
Back button doesn't always function as expected

### ISSUE THREAD

**Issue by FriederHannenheim** [moritz-weber/mucke#81]
**Is your improvement request related to a problem? Please describe.**
How to reproduce:
- Open Mucke
- Go to the library and open any artist
- Press the home button
- Press the library button
- Now press the back button
I would expect to go bock to the library and not be put on the home page

**Describe the solution you'd like**
Maybe a different back-queue for each tab (home, library and search)?

**Comment by moritz-weber** [comment]
Yeah, I know. I'm not really happy with the current implementation as well.

Right now, the "intuition" is that the back button takes you back to the previous view. This could be one of the following:
- One level up in the page stack of the current tab
- Back to the previous tab
- Both (for example, when opening an album from the home page, the tab and stack are changed in a single action)

Going back only in the same tab could be more intuitive. We can certainly try it out. Also, I want to implement that tapping on the tab icon that is already active, removes the whole stack for that tab. Example: you have a large stack of albums and artists in your library tab from browsing through your library. Tapping the library tab again, should get you back to the root (library overview).

**Comment by ildar** [comment]
completely agree. Would be great.

**Comment by marcofeltmann** [comment]
I have no clue how these changes would effect buttonless swipe-only devices like the Xiaomi Pocophone F1.

Would be worth some testing I guess.

**Comment by FriederHannenheim** [comment]
The back behavior is the same wether it's a button or a gesture. Try the above steps but replace back button with a back gesture (usually swiping from one of the sides of the phone)
